app/src/main/java/com/x8bit/bitwarden/data/vault/manager/VaultLockManagerImpl.kt

```diff
@@ -336,34 +347,25 @@ class VaultLockManagerImpl(
                         handleOnBackground()
                     }
 
-                    AppForegroundState.FOREGROUNDED -> {
-                        handleOnForeground(isFirstForeground = isFirstForeground)
-                        isFirstForeground = false
-                    }
+                    AppForegroundState.FOREGROUNDED -> handleOnForeground()
                 }
             }
             .launchIn(unconfinedScope)
     }
 
+    private fun handleOnForeground() {
+        val userId = activeUserId ?: return
+        userIdTimerJobMap[userId]?.cancel()
+    }
```

Comment:
Can we leave this function back where it was to make the diff a tiny bit smaller?